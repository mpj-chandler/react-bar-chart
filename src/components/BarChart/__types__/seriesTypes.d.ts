import DataType from "../enums/DataType";
import { Padding } from "./stylingTypes";
import { AxisRange } from "./axisTypes";
import Placement from "../enums/Placement";

export interface DataPoint {
    tag?: string;
}

export interface NamedDataPoint extends DataPoint {
    x: string;
    y: number;
}

export interface DateIndexedDataPoint extends DataPoint {
    x: Date;
    y: number;
}

export interface NumericDataPoint extends DataPoint {
    x: number | null;
    y: number | null;
}

export interface NonNullNumericDataPoint extends NumericDataPoint {
    x: number;
    y: number;
}

export interface SeriesData {
    seriesName: string;
    type: {
        x: DataType,
        y: DataType,
    };
    points: Array<NamedDataPoint | DateIndexedDataPoint | NumericDataPoint | NonNullNumericDataPoint>;
}

export interface SeriesConfig {
    numSeries: number,
    seriesIndex: number,
    padding: Padding,
    dataType: DataType,
    xRange: AxisRange,
    yRange: any,
    placement: Placement
}