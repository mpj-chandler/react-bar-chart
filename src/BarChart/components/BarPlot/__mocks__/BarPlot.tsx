import React from 'react';

const BarPlot: React.FC<any> = (props: any) => <div {...props}>Mock BarPlot {props.children}</div>

export default BarPlot;