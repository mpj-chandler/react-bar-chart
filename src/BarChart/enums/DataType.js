"use strict";
exports.__esModule = true;
var DataType;
(function (DataType) {
    DataType["Named"] = "NAMED";
    DataType["DateIndexed"] = "DATE_INDEXED";
    DataType["Numeric"] = "NUMERIC";
    DataType["NonNullNumeric"] = "NON_NULL_NUMERIC";
})(DataType || (DataType = {}));
;
exports["default"] = DataType;
