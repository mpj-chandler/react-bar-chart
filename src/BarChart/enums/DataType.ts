enum DataType {
    Named = 'NAMED',
    DateIndexed = 'DATE_INDEXED',
    Numeric = 'NUMERIC',
    NonNullNumeric = 'NON_NULL_NUMERIC',
};

export default DataType;
