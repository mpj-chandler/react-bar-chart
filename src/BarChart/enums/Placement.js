"use strict";
exports.__esModule = true;
var Placement;
(function (Placement) {
    Placement[Placement["Aligned"] = 0] = "Aligned";
    Placement[Placement["Bucket"] = 1] = "Bucket";
})(Placement || (Placement = {}));
exports["default"] = Placement;
