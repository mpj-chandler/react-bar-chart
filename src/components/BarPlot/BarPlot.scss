.BarPlot {
    grid-area: plot;
    height: 100%;
    width: 100%;
}