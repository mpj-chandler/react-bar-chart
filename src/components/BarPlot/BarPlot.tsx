import React from 'react';
import styles from './BarPlot.scss';
import Bar from '../Bar/Bar';
import { getYAxisRange } from '../../utils/axisUtils/getYAxisRange/getYAxisRange';
import { getXAxisRange } from '../../utils/axisUtils/getXAxisRange/getXAxisRange';
import { Placement } from '../../enums/Placement';
import { extractDataType } from '../../utils/axisUtils/generateAxisLabels/extractDataType';
import DataType from '../../enums/DataType';
import Axis from '../../enums/Axis';
import { Padding } from '../../__types__/styling';
import { SeriesData, NonNullNumericDataPoint, NumericDataPoint, NamedDataPoint, DateIndexedDataPoint, SeriesConfig } from '../../__types__/seriesTypes';
import { AxisConfig, AxisRange } from '../../__types__/axisTypes';
import AnimationEasingType from '../../enums/AnimationEasingFunction';

export interface BarPlotProps {
    padding: Padding;
    data: SeriesData[];
    xAxisConfig?: AxisConfig;
    yAxisConfig?: AxisConfig;
    animationEasingFunction: AnimationEasingType;
}

const BarPlot: React.FC<BarPlotProps> = (props: BarPlotProps) => {
    const colors = ['#283593', '#0277bd', '#2e7d32', '#c62828'];

    const renderSeries: (seriesData: SeriesData, config: SeriesConfig) => React.ReactNode = (seriesData, config) => {
        return seriesData.points.map(
            (
                point: NonNullNumericDataPoint | NumericDataPoint | NamedDataPoint | DateIndexedDataPoint,
                index: number,
            ) => {
            if (point.x !== null && point.y !== null) {
                return (
                    <Bar
                        key={`${seriesData.seriesName}${point.x}`}
                        point={point}
                        { ...config }
                        width={10}
                        fill={colors[config.seriesIndex % colors.length]}
                        stroke={'black'}
                        numBars={seriesData.points.length}
                        placement={placement}
                        index={index}
                        animationEasingType={props.animationEasingFunction}
                    />
                );
            }

            return null;
        });
    }


    const xRange = getXAxisRange(props.data, props.xAxisConfig);
    const yRange = getYAxisRange(props.data, props.yAxisConfig);
    const placement = props.xAxisConfig ? props.xAxisConfig.tickPlacement : Placement.Aligned;
    const dataType: DataType = extractDataType(props.data, Axis.XAxis);

    return (
        <svg className={styles.BarPlot}>
            <g>
                {props.data.map((seriesData: SeriesData, index: number) => {
                    const seriesConfig: SeriesConfig = {
                        numSeries: props.data.length,
                        seriesIndex: index,
                        padding: props.padding,
                        dataType,
                        xRange,
                        yRange,
                        placement,
                    };

                    return renderSeries(seriesData, seriesConfig);
                })}
            </g>
        </svg>
    );
}

export default BarPlot;

