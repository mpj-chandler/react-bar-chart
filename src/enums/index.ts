import Axis from './Axis';
import DataType from './DataType';
import { Placement } from './Placement';

export {
    Axis,
    DataType,
    Placement
}