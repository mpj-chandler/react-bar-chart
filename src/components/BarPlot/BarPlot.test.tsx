import React from 'react';
import TestRenderer from 'react-test-renderer';
import BarPlot, { BarPlotProps } from './BarPlot';

import { Placement } from '../../enums/Placement';
import DataType from '../../enums/DataType';
import { AxisConfig } from '../../__types__/axisTypes';

jest.mock('../Bar/Bar.tsx');

describe('BarPlot', () => {
    const inlineConfig: AxisConfig = {
        zeroIntercept: false,
        margin: 0,
        tickPlacement: Placement.Aligned,
        tickLength: 100,
    };

    const bucketConfig: AxisConfig = { ...inlineConfig, tickPlacement: Placement.Bucket };

    const props: BarPlotProps = {
        data: [
            {
                seriesName: 'A',
                type: {
                    x: DataType.NonNullNumeric,
                    y: DataType.NonNullNumeric,
                },
                points: [
                    {
                        x: 0,
                        y: 10,
                    },
                    {
                        x: 1,
                        y: 20,
                    },
                    {
                        x: 2,
                        y: 30,
                    }
                ],
            },
        ],
        padding: {
            top: 10,
            bottom: 10,
            left: 10,
            right: 10,
        },
    };

    describe('When all data points are non-null', () => {

        it('it renders consistently with defaults', () => {
            const component = TestRenderer.create(<BarPlot {...props}/>);

            expect(component.toJSON()).toMatchSnapshot();
        });

        it('it renders consistently with tick alignment explicitly set to Aligned for both axes', () => {
            const component = TestRenderer.create(
                <BarPlot
                    {...props}
                    xAxisConfig={inlineConfig}
                    yAxisConfig={inlineConfig}
                />);

            expect(component.toJSON()).toMatchSnapshot();
        });

        it('it renders consistently with tick alignment explicitly set to Aligned for XAxis and Bucket for YAxis',
        () => {
            const component = TestRenderer.create(
                <BarPlot
                    {...props}
                    xAxisConfig={inlineConfig}
                    yAxisConfig={bucketConfig}
                />);

            expect(component.toJSON()).toMatchSnapshot();
        });

        it('it renders consistently with tick alignment explicitly set to Bucket for XAxis and Aligned for YAxis',
        () => {
            const component = TestRenderer.create(
                <BarPlot
                    {...props}
                    xAxisConfig={bucketConfig}
                    yAxisConfig={inlineConfig}
                />);

            expect(component.toJSON()).toMatchSnapshot();
        });

        it('it renders consistently with tick alignment explicitly set to Bucket for both axes',
        () => {
            const component = TestRenderer.create(
                <BarPlot
                    {...props}
                    xAxisConfig={bucketConfig}
                    yAxisConfig={bucketConfig}
                />);

            expect(component.toJSON()).toMatchSnapshot();
        });
    });

    describe('When some data points are null', () => {

        const patchyProps = {
            ...props,
            data: [
                {
                    seriesName: 'A',
                    type: {
                        x: DataType.NonNullNumeric,
                        y: DataType.NonNullNumeric,
                    },
                    points: [
                        {
                            x: null,
                            y: 10,
                        },
                        {
                            x: 1,
                            y: null,
                        },
                        {
                            x: 2,
                            y: 30,
                        }
                    ],
                },
            ],
        };

        it('it renders consistently with defaults', () => {
            const component = TestRenderer.create(<BarPlot {...patchyProps}/>);

            expect(component.toJSON()).toMatchSnapshot();
        });

        it('it renders consistently with tick alignment explicitly set to Aligned for both axes', () => {
            const component = TestRenderer.create(
                <BarPlot
                    {...patchyProps}
                    xAxisConfig={inlineConfig}
                    yAxisConfig={inlineConfig}
                />);

            expect(component.toJSON()).toMatchSnapshot();
        });

        it('it renders consistently with tick alignment explicitly set to Aligned for XAxis and Bucket for YAxis',
        () => {
            const component = TestRenderer.create(
                <BarPlot
                    {...patchyProps}
                    xAxisConfig={inlineConfig}
                    yAxisConfig={bucketConfig}
                />);

            expect(component.toJSON()).toMatchSnapshot();
        });

        it('it renders consistently with tick alignment explicitly set to Bucket for XAxis and Aligned for YAxis',
        () => {
            const component = TestRenderer.create(
                <BarPlot
                    {...patchyProps}
                    xAxisConfig={bucketConfig}
                    yAxisConfig={inlineConfig}
                />);

            expect(component.toJSON()).toMatchSnapshot();
        });

        it('it renders consistently with tick alignment explicitly set to Bucket for both axes',
        () => {
            const component = TestRenderer.create(
                <BarPlot
                    {...patchyProps}
                    xAxisConfig={bucketConfig}
                    yAxisConfig={bucketConfig}
                />);

            expect(component.toJSON()).toMatchSnapshot();
        });
    });
})
