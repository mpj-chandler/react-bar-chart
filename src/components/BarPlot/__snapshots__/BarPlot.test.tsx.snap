// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarPlot When all data points are non-null it renders consistently with defaults 1`] = `
<svg
  className="BarPlot"
>
  <g>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={0}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 0,
          "y": 10,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={1}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 1,
          "y": 20,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={2}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 2,
          "y": 30,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
  </g>
</svg>
`;

exports[`BarPlot When all data points are non-null it renders consistently with tick alignment explicitly set to Aligned for XAxis and Bucket for YAxis 1`] = `
<svg
  className="BarPlot"
>
  <g>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={0}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 0,
          "y": 10,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={1}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 1,
          "y": 20,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={2}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 2,
          "y": 30,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
  </g>
</svg>
`;

exports[`BarPlot When all data points are non-null it renders consistently with tick alignment explicitly set to Aligned for both axes 1`] = `
<svg
  className="BarPlot"
>
  <g>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={0}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 0,
          "y": 10,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={1}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 1,
          "y": 20,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={2}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 2,
          "y": 30,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
  </g>
</svg>
`;

exports[`BarPlot When all data points are non-null it renders consistently with tick alignment explicitly set to Bucket for XAxis and Aligned for YAxis 1`] = `
<svg
  className="BarPlot"
>
  <g>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={0}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={1}
      point={
        Object {
          "x": 0,
          "y": 10,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2.6666666666666665,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={1}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={1}
      point={
        Object {
          "x": 1,
          "y": 20,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2.6666666666666665,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={2}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={1}
      point={
        Object {
          "x": 2,
          "y": 30,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2.6666666666666665,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
  </g>
</svg>
`;

exports[`BarPlot When all data points are non-null it renders consistently with tick alignment explicitly set to Bucket for both axes 1`] = `
<svg
  className="BarPlot"
>
  <g>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={0}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={1}
      point={
        Object {
          "x": 0,
          "y": 10,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2.6666666666666665,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={1}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={1}
      point={
        Object {
          "x": 1,
          "y": 20,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2.6666666666666665,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={2}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={1}
      point={
        Object {
          "x": 2,
          "y": 30,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2.6666666666666665,
          "min": 0,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
  </g>
</svg>
`;

exports[`BarPlot When some data points are null it renders consistently with defaults 1`] = `
<svg
  className="BarPlot"
>
  <g>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={2}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 2,
          "y": 30,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 1,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
  </g>
</svg>
`;

exports[`BarPlot When some data points are null it renders consistently with tick alignment explicitly set to Aligned for XAxis and Bucket for YAxis 1`] = `
<svg
  className="BarPlot"
>
  <g>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={2}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 2,
          "y": 30,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 1,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
  </g>
</svg>
`;

exports[`BarPlot When some data points are null it renders consistently with tick alignment explicitly set to Aligned for both axes 1`] = `
<svg
  className="BarPlot"
>
  <g>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={2}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={0}
      point={
        Object {
          "x": 2,
          "y": 30,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2,
          "min": 1,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
  </g>
</svg>
`;

exports[`BarPlot When some data points are null it renders consistently with tick alignment explicitly set to Bucket for XAxis and Aligned for YAxis 1`] = `
<svg
  className="BarPlot"
>
  <g>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={2}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={1}
      point={
        Object {
          "x": 2,
          "y": 30,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2.3333333333333335,
          "min": 1,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
  </g>
</svg>
`;

exports[`BarPlot When some data points are null it renders consistently with tick alignment explicitly set to Bucket for both axes 1`] = `
<svg
  className="BarPlot"
>
  <g>
    <div
      animationEasingType="none"
      dataType="NON_NULL_NUMERIC"
      fillFormatter={[Function]}
      index={2}
      numBars={3}
      numSeries={1}
      padding={
        Object {
          "bottom": 10,
          "left": 10,
          "right": 10,
          "top": 10,
        }
      }
      placement={1}
      point={
        Object {
          "x": 2,
          "y": 30,
        }
      }
      seriesIndex={0}
      stroke="black"
      width={10}
      xRange={
        Object {
          "max": 2.3333333333333335,
          "min": 1,
        }
      }
      yRange={
        Object {
          "max": 30,
          "min": 10,
        }
      }
    >
      Mock Bar 
    </div>
  </g>
</svg>
`;
